Objects/bytearrayobject.c

```diff
@@ -709,7 +709,9 @@ bytearray_ass_subscript_lock_held(PyObject *op, PyObject *index, PyObject *value
     _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(op);
     PyByteArrayObject *self = _PyByteArray_CAST(op);
     Py_ssize_t start, stop, step, slicelen;
-    char *buf = PyByteArray_AS_STRING(self);
+    // Do not store a reference to the internal buffer since
+    // index.__index__() or _getbytevalue() may alter 'self'.
+    // See https://github.com/python/cpython/issues/91153.
```

Comment:
I don't think we need to fully link to it, just the `gh-` reference should be fine.
```suggestion
    // See gh-91153
```

Reply:
picnixz suggested this above, and it's common for the c source to have complete links, so I think it's fine

Reply:
With complete links, I can open them from my IDE. It's much easier that way IMO